### ISSUE TITLE
Webjob only executes if triggered manually, but shows success with timed trigger without execution

### ISSUE THREAD

**Issue by sweetmoniker** [projectkudu/kudu#2768]
First, the app service is set to "Always On."

I have a webjob that is culling data from Google Search Console and writing it to a SQL Database in Azure. It works great if I go into the Webjobs dashboard and hit "Run." I set up a CRON trigger for 004***. I can see the webjob getting triggered in the logs, but there is no execution - nothing goes into my database. Additionally, WebJob Details shows "Success" every time. 

![image](https://user-images.githubusercontent.com/29126278/40022885-453ea8aa-5797-11e8-9f2b-e7485c216a13.png)

The jobs that are executed are when I manually hit "Run" in the dashboard. These are the 3 runs with a duration of 5 minutes. The three runs with 23, 29, and 35 seconds as far as I can tell didn't actually execute anything. 

job_scheduler.log shows the following:
```
[05/10/2018 21:43:53 > 2bb9db: SYS INFO] Scheduling WebJob with 0 0 4 * * * next schedule expected in 06:16:06.0532568 
[05/11/2018 13:37:18 > 2bb9db: SYS WARN] Missed 1 schedules, most recent at 5/11/2018 4:00:00 AM
[05/11/2018 13:37:18 > 2bb9db: SYS INFO] Scheduling WebJob with 0 0 4 * * * next schedule expected in 14:22:41.2448358
[05/11/2018 13:39:34 > 2bb9db: SYS WARN] Missed 1 schedules, most recent at 5/11/2018 4:00:00 AM
[05/11/2018 13:39:34 > 2bb9db: SYS INFO] Scheduling WebJob with 0 0 4 * * * next schedule expected in 14:20:25.5168678
[05/12/2018 03:59:59 > 2bb9db: SYS INFO] Next schedule expected in 00:00:00.4010097
[05/12/2018 04:00:00 > 2bb9db: SYS INFO] WebJob invoked
[05/12/2018 04:00:00 > 2bb9db: SYS INFO] Next schedule expected in 23:59:59.6296632
[05/13/2018 03:59:58 > 2bb9db: SYS INFO] Next schedule expected in 00:00:01.2438851
[05/13/2018 04:00:01 > 2bb9db: SYS INFO] WebJob invoked
[05/13/2018 04:00:01 > 2bb9db: SYS INFO] Next schedule expected in 23:59:58.9957291
[05/14/2018 03:59:59 > 2bb9db: SYS INFO] Next schedule expected in 00:00:00.8204240
[05/14/2018 04:00:01 > 2bb9db: SYS INFO] WebJob invoked
[05/14/2018 04:00:01 > 2bb9db: SYS INFO] Next schedule expected in 23:59:58.0301390
[05/14/2018 15:14:34 > 2bb9db: SYS INFO] Scheduling WebJob with 0 0 4 * * * next schedule expected in 12:45:25.0708277
```

The status file from today's manual run:
`{"Trigger":"External - Portal-Exp/5.12.33.178 (Websites) Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML%2C like Gecko) Chrome/66.0.3359.139 Safari/537.36","Status":"Success","StartTime":"2018-05-14T15:07:01.4637988Z","EndTime":"2018-05-14T15:12:23.5108655Z"}`

The status file from today's scheduled run:
`{"Trigger":"Schedule - 0 0 4 * * *","Status":"Success","StartTime":"2018-05-14T04:00:01.1909443Z","EndTime":"2018-05-14T04:00:36.5704235Z"}`

The output_log.txt from today's manual run:
```
[05/14/2018 15:07:01 > 2bb9db: SYS INFO] Status changed to Initializing
[05/14/2018 15:07:01 > 2bb9db: SYS INFO] Job directory change detected: Job file 'webmasters.dat' timestamp differs between source and working directories.
[05/14/2018 15:07:04 > 2bb9db: SYS INFO] Run script 'run.cmd' with script host - 'WindowsScriptHost'
[05/14/2018 15:07:04 > 2bb9db: SYS INFO] Status changed to Running
[05/14/2018 15:07:04 > 2bb9db: INFO] 
[05/14/2018 15:07:04 > 2bb9db: INFO] D:\local\Temp\jobs\triggered\SearchConsole\lug03uat.fje>D:\Python34\python.exe GoogleSearchConsole_Azure.py 
[05/14/2018 15:12:23 > 2bb9db: SYS INFO] Status changed to Success
```

The output_log.txt from today's scheduled run:
```
[05/14/2018 04:00:01 > 2bb9db: SYS INFO] Status changed to Initializing
[05/14/2018 04:00:04 > 2bb9db: SYS INFO] Job directory change detected: Job file 'webmasters.dat' timestamp differs between source and working directories.
[05/14/2018 04:00:15 > 2bb9db: SYS INFO] Run script 'run.cmd' with script host - 'WindowsScriptHost'
[05/14/2018 04:00:15 > 2bb9db: SYS INFO] Status changed to Running
[05/14/2018 04:00:16 > 2bb9db: INFO] 
[05/14/2018 04:00:16 > 2bb9db: INFO] D:\local\Temp\jobs\triggered\SearchConsole\qbzjk5ev.ghi>D:\Python34\python.exe GoogleSearchConsole_Azure.py 
[05/14/2018 04:00:36 > 2bb9db: SYS INFO] Status changed to Success
```

Notice the time differences between the scheduled and manual runs. 35 seconds is for sure not enough time to actually execute the script. What the webjob is doing in those 23 to 35 seconds I am unsure, but there are no visible manifestations that I can see. Any thoughts? What am I missing?

**Comment by davidebbo** [comment]
Not sure if it's related, but one message that looks puzzling is:

    Job directory change detected: Job file 'webmasters.dat' timestamp differs between source and working directories.

Normally, this should only happen if you are modifying the content of your WebJob folder. Do you think that is in fact happening?

Also, have you tried adding some log lines at the very beginning/end of the Python code, so see if they're there in the scheduled case?

**Comment by sweetmoniker** [comment]
Thanks for the response! 

I noticed the webmasters.dat line. Google credentials are stored in that file; however, I just checked Kudu, and the time stamp is the original creation date. In fact, all files have the time stamp of their creation date. Additionally, that line occurs whether the Webjob is run manually or scheduled.

I have not tried adding log code. To be honest, I didn't know that was possible. I will test and get back to you. 

Thanks again.

**Comment by davidebbo** [comment]
@sweetmoniker that message could be unrelated, though it is strange that it detects a change. Let's see if you can find anything with logging.

**Comment by sweetmoniker** [comment]
As I was adding logging code, I realized what the problem was likely to be. Given the way the code was written and the dates of data being requested, and since the trigger was scheduled for 4AM, it was before data had been posted by Google. When I requested the data manually, later in the day, it would work fine because the data would be available. So the script was executing correctly. There was just nothing to process. Adjusting the dates being requested (and the trigger time) worked flawlessly. I guess we can chalk this one up to "user error." My apologies. I wish I had thought about it a bit harder before posting. 

Additionally, and somewhat bizarrely, the webmasters.dat line is not in the most recent output_log.txt. Why that is I am unsure, as I did nothing that should affect script behavior with regards to that file.

At any rate, thanks for your time.
